### PR TITLE
Disable `PageIcon`, `InsertBlock` & `PageTitle` on readonly mode

### PR DESCRIPTION
**Pull request by yusufkinatas**
## 🌟 What is the purpose of this PR?

Disable following elements on pages on `readonly` mode, to prevent the users from seeing editable elements on readonly pages. 

- Page title
- Page icon
- Insert block element (full-width horizontal element between blocks)

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1200211978612931/1202855617846688/f) _(internal)_

## ❓ How to test this?

1.  Checkout the branch / view the deployment
1.  Open a page and add `&readonly` as url param
1.  Confirm that elements mentioned above are not available.

### CODE REVIEW

File: packages/hash/frontend/src/blocks/page/PageTitle/PageTitle.tsx [hunk 1]
```diff
@@ -107,7 +110,7 @@ export const PageTitle: FunctionComponent<PageTitleProps> = ({
     <StyledTextarea
       ref={pageTitleRef}
       placeholder={PAGE_TITLE_PLACEHOLDER}
-      disabled={updateEntityLoading}
+      disabled={updateEntityLoading || readonlyMode}
```
Comment (luisbettencourt) [inline packages/hash/frontend/src/blocks/page/PageTitle/PageTitle.tsx:113]:
```suggestion
      readOnly={readOnlyMode}
      disabled={updateEntityLoading}
```

File: packages/hash/frontend/src/blocks/page/PageTitle/PageTitle.tsx [hunk 2]
```diff
@@ -24,14 +25,16 @@ const StyledTextarea = styled(TextareaAutosize)(({ theme }) =>
...
     lineHeight: PAGE_TITLE_LINE_HEIGHT,
     fontWeight: 500,
+    color: theme.palette.black,
 
     "&::placeholder": {
       color: theme.palette.gray[40],
       opacity: 1,
     },
 
     ":disabled": {
```
Comment (luisbettencourt) [inline packages/hash/frontend/src/blocks/page/PageTitle/PageTitle.tsx:35]: We could revert this if we go with `readOnly` insetead of `disabled`.

**Review by luisbettencourt — CHANGES_REQUESTED** [review]
Looks good, just a minor suggestion that I think would slightly simplify this. Let me know what you think!

File: packages/hash/frontend/src/blocks/page/PageTitle/PageTitle.tsx (hunk 1 above)
Comment (yusufkinatas) [reply packages/hash/frontend/src/blocks/page/PageTitle/PageTitle.tsx:113]: Hmm, a `disabled` text area means "non editable" text area right? So don't want users to edit the text area when loading, and when it's on readonly mode. So I feel like current implementation makes sense and simple enough. Let me know if you have another concern regarding this.

File: packages/hash/frontend/src/blocks/page/PageTitle/PageTitle.tsx (hunk 2 above)
Comment (yusufkinatas) [reply packages/hash/frontend/src/blocks/page/PageTitle/PageTitle.tsx:35]: putting `color` here prevented `.disabled` to have a different color.

Comment (luisbettencourt) [reply packages/hash/frontend/src/blocks/page/PageTitle/PageTitle.tsx:35]: I meant the disabled `:disabled` styles, not the color. Sorry the comment placement wasn't very clear

**Review by luisbettencourt — APPROVED** [review]

**Comment by yusufkinatas** [comment]
Thanks for the review @luisbettencourt!

**Pull request merged**